src/bm_sim/actions.cpp

```diff
@@ -282,6 +283,25 @@ ActionFn::parameter_end_vector() {
   params.back().params_vector.end = end;
 }
 
+void
+ActionFn::parameter_start_field_list() {
+  ActionParam param;
+  param.tag = ActionParam::FIELD_LIST;
+  auto start = static_cast<unsigned int>(field_params.size());
+  param.field_list = {start, start /* end */};
+  params.push_back(param);
+  params.swap(field_params);
+}
+
+void
+ActionFn::parameter_end_field_list() {
+  params.swap(field_params);
+  assert(params.back().tag == ActionParam::FIELD_LIST &&
+         "no field list was started");
+  auto end = static_cast<unsigned int>(field_params.size());
+  params.back().field_list.end = end;
+}
```

Comment:
it feels like we could have used `sub_params` for this. Did we need a new vector (`field_params`)?